corehq/apps/locations/tests/test_bulk_management.py

```diff
@@ -0,0 +1,113 @@
+from django.test import TestCase
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.locations.bulk_management import bulk_update_organization
+
+# These example types and trees mirror the information available in the upload files
+
+# TODO What fields should be editable in the upload?
+# Look at what's editable in the UI, especially with commtrack enabled versus disabled
+
+FLAT_LOCATION_TYPES = [
+    # ('name', 'code', 'parent_code', 'shares_cases', 'view_descendants'),
+    ('State', 'state', '', False, False),
+    ('County', 'county', 'state', False, True),
+    ('City', 'city', 'county', True, False),
+]
+BASIC_LOCATION_TREE = [
+    # ('name', 'site_code', 'location_type', 'parent_code', 'location_id',
+    # 'external_id', 'latitude', 'longitude'),
+    ('Massachusetts', 'mass', 'state', '', '1234', '', '', ''),
+    ('Suffolk', 'suffolk', 'county', 'mass', '2345', '', '', ''),
+    ('Boston', 'boston', 'city', 'suffolk', '2346', '', '', ''),
+    ('Middlesex', 'middlesex', 'county', 'mass', '3456', '', '', ''),
+    ('Cambridge', 'cambridge', 'city', 'middlesex', '3457', '', '', ''),
+    ('Florida', 'florida', 'state', '', '5432', '', '', ''),
+    ('Duval', 'duval', 'county', 'florida', '5433', '', '', ''),
+    ('Jacksonville', 'jacksonville', 'city', 'duval', '5434', '', '', ''),
+]
+
+MOVE_SUFFOLK_TO_FLORIDA = [
+    # ('name', 'site_code', 'location_type', 'parent_code', 'location_id',
+    # 'external_id', 'latitude', 'longitude'),
+    ('Massachusetts', 'mass', 'state', '', '1234', '', '', ''),
+    # this is the only changed line (parent is changed to florida)
+    ('Suffolk', 'suffolk', 'county', 'florida', '2345', '', '', ''),
+    ('Boston', 'boston', 'city', 'suffolk', '2346', '', '', ''),
+    ('Middlesex', 'middlesex', 'county', 'mass', '3456', '', '', ''),
+    ('Cambridge', 'cambridge', 'city', 'middlesex', '3457', '', '', ''),
+    ('Florida', 'florida', 'state', '', '5432', '', '', ''),
+    ('Duval', 'duval', 'county', 'florida', '5433', '', '', ''),
+    ('Jacksonville', 'jacksonville', 'city', 'duval', '5434', '', '', ''),
+]
+
+DELETE_SUFFOLK = [
+    # ('name', 'site_code', 'location_type', 'parent_code', 'location_id',
+    # 'external_id', 'latitude', 'longitude', 'delete'),
+    ('Massachusetts', 'mass', 'state', '', '1234', '', '', '', ''),
+    # These next two are marked as 'delete'
+    ('Suffolk', 'suffolk', 'county', 'mass', '2345', '', '', '', 'delete'),
+    ('Boston', 'boston', 'city', 'suffolk', '2346', '', '', '', 'delete'),
+    ('Middlesex', 'middlesex', 'county', 'mass', '3456', '', '', '', ''),
+    ('Cambridge', 'cambridge', 'city', 'middlesex', '3457', '', '', '', ''),
+    ('Florida', 'florida', 'state', '', '5432', '', '', '', ''),
+    ('Duval', 'duval', 'county', 'florida', '5433', '', '', '', ''),
+    ('Jacksonville', 'jacksonville', 'city', 'duval', '5434', '', '', '', ''),
+]
```

Comment:
These could also be represented as CSVs if that's easier.  I don't have a strong preference there.  @sravfeyn what would you prefer?